File: 2315. Count Asterisks.rb
#* 2315. Count Asterisks
# s = "iamprogrammer"
s = "yo|uar|e**|b|e***au|tifu|l"
# @param {String} s
# @return {Integer}
#* SOL 1.1
def count_asterisks(s)
    res=0
    s.split('|').each_with_index do |subs, i|
        res += subs.count('*') if i.even?
    end
    res  
end

#*
def count_asterisks(s)
    res=[]
    s.split('|').each_with_index do |subs, i|
        p "#{subs} #{i.even?} #{subs.count('*')}"
        res << subs.count('*') if i.even?
    end
    res.sum   
end
p count_asterisks(s)

#*WORKING . . .
#*DONT WORK
#*
def count_asterisks(s)
    s.split('|').each_with_index.count do |sub, i|
        sub.count('*') if i.even?
    end    
end
p count_asterisks(s)

#*
def count_asterisks(s)
    s.count('|') == 0 ? 0 : s.split('|').each_with_index.count{ |sub, i|sub.count('*') if i.even? }    
end
p count_asterisks(s)

File: 2315. Count Asterisks.py
#* 2315. Count Asterisks
# s = "iamprogrammer"
s = "yo|uar|e**|b|e***au|tifu|l"
# @param {String} s
# @return {Integer}
#* SOL 1.1 | NO PUTS
class Solution:
    def countAsterisks(self, s: str) -> int:
        res=0
        x = s.split('|')

        for i, substr in enumerate(x):
            if i % 2 == 0:
                res += substr.count('*')

        return res
    
print(Solution().countAsterisks(s))
    
#* SOL 1.1
def countAsterisks(s):
  res=0
  x = s.split('|')
  print(x)

  for i, substr in enumerate(x):
    if i % 2 == 0:
        print(substr.count('*'))
        res += substr.count('*')

  return res

print(countAsterisks(s))


#*WORKING. . .
#* NO GOOD
def countAsterisks(s):
  res=0
  x = s.split('|')
  print(x)

  for i, substr in enumerate(x):
    print(substr)
    if i % 2 == 0:
      res += substr.count('*')

    return res
  
s = "yo|uar|e**|b|e***au|tifu|l"
print(countAsterisks(s))
